### DISCUSSION TITLE
Cartridge aromatization of SMARTS queries

### DISCUSSION THREAD

**Discussion by someonetookmyname** [rdkit/rdkit#7207] [Q&A] [answered]
The cartridge documentation (https://www.rdkit.org/docs/Cartridge.html) indicates that mol_adjust_query_properties() will perform aromaticity perception, but I cannot get a kekule bezene SMARTS to find either aromatic benzene or kekule benzene (both sanitized to aromatic) by substructure search.  Aromatic benzene SMARTS works fine.  See below for both.

Is there a way to get the kekule benzene SMARTS to work the same as aromatic benzene SMARTS?

Thank you for any guidance.

Best, Andrew

```
create table test (pk integer, m mol);
create index testidx on test using gist(m);
insert into test (pk, m) values (1, 'c1ccccc1'::mol);
insert into test (pk, m) values (2, 'C1=CC=CC=C1'::mol);

select pk, m from test where m @> mol_adjust_query_properties('c1ccccc1'::qmol);
 pk |    m     
----+----------
  1 | c1ccccc1
  2 | c1ccccc1
(2 rows)

select pk, m from test where m @> mol_adjust_query_properties('[#6]1=[#6]-[#6]=[#6]-[#6]=[#6]-1'::qmol);
 pk | m 
----+---
(0 rows)
```

**Comment by greglandrum** [comment, marked as answer]
@someonetookmyname Thearomaticity perception done by `mol_adjust_query_properties()` does not, as you have discovered, work with molecules constructed from SMARTS. This isn't something we are likely to ever change.

SMARTS is a language which gives you enormous control over how to express your queries, that includes specifying exactly what bond orders you want. If you do not what this degree of specificity, you should provide the query molecules as SMILES or a mol block (CTAB).

**Reply by someonetookmyname** [reply]
Thank you for the response and clarification.  Very helpful.
